src/pmdas/rsyslog/pmdarsyslog.pl

```diff
@@ -27,34 +27,149 @@
 my @queue_insts = ();
 use vars qw(%queue_ids %queue_values);
 
-# .* rsyslogd-pstats:
-# imuxsock: submitted=37 ratelimit.discarded=0 ratelimit.numratelimiters=22 
-# elasticsearch: connfail=0 submits=0 failed=0 success=0 
-# [main Q]: size=1 enqueued=1436 full=0 maxqsize=3 
+# Legacy method
+#   .* rsyslogd-pstats:
+#   imuxsock: submitted=37 ratelimit.discarded=0 ratelimit.numratelimiters=22
+#   elasticsearch: connfail=0 submits=0 failed=0 success=0
+#   [main Q]: size=1 enqueued=1436 full=0 maxqsize=3
+
+# Modern method
+#   module(load="imuxsock")
+#   input(type="imuxsock"
+#       Socket="/dev/log"
+#       CreatePath="on"
+#   )
+#   module(load="impstats" interval="10" severity="7")
+#   if $syslogtag contains 'rsyslogd-pstats' then {
+#       action(
+#           name="pstats-for-pcp"
+#           type="omfile"
+#           template="RSYSLOG_FileFormat"
+#           file="/var/log/pcp/rsyslog/stats"
+#       )
+#       stop
+#   }
```

Comment:
The man page for pmdarsyslog should document this modern method now too I guess, in the INSTALLATION section.  (src/pmdas/rsyslog/pmdarsyslog.1)

Reply:
Addressed in latest update.